volatility3/framework/plugins/windows/thrdscan.py

```diff
@@ -67,39 +67,109 @@ def scan_threads(
             yield mem_object
 
     @classmethod
-    def gather_thread_info(cls, ethread):
+    def gather_thread_info(
+        cls,
+        ethread: interfaces.objects.ObjectInterface,
+        vads_cache: Dict[int, pe_symbols.ranges_type] = None,
+    ) -> Tuple[
+        int,
+        int,
+        int,
+        int,
+        Optional[str],
+        int,
+        Optional[str],
+        Optional[datetime.datetime],
+        Optional[datetime.datetime],
+    ]:
         try:
             thread_offset = ethread.vol.offset
             owner_proc_pid = ethread.Cid.UniqueProcess
             thread_tid = ethread.Cid.UniqueThread
             thread_start_addr = ethread.StartAddress
+            thread_win32start_addr = ethread.Win32StartAddress
             thread_create_time = (
                 ethread.get_create_time()
             )  # datetime.datetime object / volatility3.framework.renderers.UnparsableValue object
             thread_exit_time = (
                 ethread.get_exit_time()
             )  # datetime.datetime object / volatility3.framework.renderers.UnparsableValue object
+
+            owner_proc = None
+            if vads_cache is not None:
+                owner_proc = ethread.owning_process()
         except exceptions.InvalidAddressException:
             vollog.debug(f"Thread invalid address {ethread.vol.offset:#x}")
             return None
 
+        if vads_cache is not None:
+            vads = pe_symbols.PESymbols.get_vads_for_process_cache(
+                vads_cache, owner_proc
+            )
+            # no vads = terminated/smeared, pid 4 = kernel = don't check VADs
+            if (
+                owner_proc_pid != 4
+                and owner_proc.InheritedFromUniqueProcessId != 4
```

Comment:
Do we need to check all the parents, or will the one level of InheritedFrom be enough?

Reply:
Just one level is good. Kernel process (Registry, MemCompression, etc.) will always have a parent of 4